backend/wayland/backend.c

```diff
@@ -225,7 +225,8 @@ static void registry_global(void *data, struct wl_registry *registry,
 			&zxdg_decoration_manager_v1_interface, 1);
 	} else if (strcmp(iface, zwp_pointer_gestures_v1_interface.name) == 0) {
 		wl->zwp_pointer_gestures_v1 = wl_registry_bind(registry, name,
-			&zwp_pointer_gestures_v1_interface, 1);
+			&zwp_pointer_gestures_v1_interface,
+			fmin(zwp_pointer_gestures_v1_interface.version, 3));
```

Comment:
`zwp_pointer_gestures_v1_interface.version` is the version the header was generated with, not the version supported by the parent compositor. The `version` variable actually indicates the version supported by the parent compositor.

If possible, I'd prefer to avoid using `fmin`, I'm worried about implicit number conversion rules and rounding shenanigans. `version < 3 ? version : 3` would work.

Reply:
Oooops I hope this time I got it right. Tanks for your patience!